src/bootstrap/compile.rs

```diff
@@ -553,6 +553,18 @@ pub fn rustc_cargo_env(builder: &Builder, cargo: &mut Command) {
     if builder.config.rustc_parallel_queries {
         cargo.env("RUSTC_PARALLEL_QUERIES", "1");
     }
+
+    let use_polly = match builder.kind {
```

Comment:
`builder.kind` only records how `x.py` is invoked, i.e. if you use 

```sh
./x.py test src/test/compile-fail ...
```

then librustc etc will be built with `builder.config.rust_polly_tests` instead of `builder.config.rust_polly_self`, even if it is a build artifact.